File: miraculous-frontend/miraculous-ui/src/dto/user-settings-dto/user-settings-dto.ts
export class UserSettingsDto {

  id: number;
  email: string;
  description: string;
  location: string;
  coverPictureUrlName: string;
  password: string;
  confirmationPassword: string;


  constructor(id?: number, email?: string,  description?: string, location?: string, coverPictureUrlName?: string,
    password?: string, confirmationPassword?: string) {
    this.id = id;
    this.email = email;
    this.description = description;
    this.location = location;
    this.coverPictureUrlName = coverPictureUrlName;
    this.password = password;
    this.confirmationPassword = confirmationPassword;
  }
}

File: miraculous-backend/src/main/java/com/example/miraculousbackend/dto/UserSettingsDto.java
package com.example.miraculousbackend.dto;

import com.example.miraculousbackend.validation.FieldMatch;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldMatch(message = "Passwords do not match", firstField = "password", secondField = "confirmationPassword")
public class UserSettingsDto {

    private Long id;
    @NotBlank(message = "You must provide an email")
    @Email(message = "You must enter a valid email")
    private String email;
    private String description;
    private String location;
    @Size(min = 6, message = "Your password must have at least 6 symbols")
    private String password;
    @Size(message = "Confirmation password must have at least 6 symbols", min = 6)
    private String confirmationPassword;
    private String coverPictureUrlName;

}
